src/malli/dev/cljs.cljc

```diff
@@ -17,6 +18,8 @@
    (defn start!* [env options]
      `(do
         ;; register all function schemas and instrument them based on the options
+        ;; first clear out all metadata schemas to support dev-time removal of metadata schemas on functions - they should not be instrumented
+        ~(m/-deregister-metadata-function-schemas! :cljs)
```

Comment:
I think it is not needed on `stop!` because there are two separate concerns: 
- collecting the functions that have schemas and storing them in a hashmap in an atom
- starting and stopping instrumentation

the `-deregister` helper is to deal with ensuring those functions stored in the atom are up to date with the code, whereas `stop!`ing is only concerned with replacing the instrumented function with its original function.

I could see some weird interaction happening with hot reloading where you both remove the schema and change the function implementation, and then call `stop!` which may replace the updated function with the original function.
But this would only affect one reloaded app instance, you would just have to hot-reload again and the newest function will be in place.

I tried out running `stop!` after removing metadata and things work as expected, but of course if you run into something strange let me know.